mllib/src/main/scala/org/apache/spark/ml/classification/LinearSVC.scala

```diff
@@ -0,0 +1,525 @@
+      val bcFeaturesStd = instances.context.broadcast(featuresStd)
+      val costFun = new LinearSVCCostFun(instances, numClasses, $(fitIntercept),
+        $(standardization), bcFeaturesStd, regParamL2, $(aggregationDepth))
+
+      def regParamL1Fun = (index: Int) => 0D
+      val optimizer = new BreezeOWLQN[Int, BDV[Double]]($(maxIter), 10, regParamL1Fun, $(tol))
+      val initialCoefficientsWithIntercept = Vectors.zeros(numFeaturesPlusIntercept)
+      if ($(fitIntercept)) {
+        initialCoefficientsWithIntercept.toArray(numFeatures) = math.log(
+          histogram(1) / histogram(0))
+      }
+
+      val states = optimizer.iterations(new CachedDiffFunction(costFun),
+        initialCoefficientsWithIntercept.asBreeze.toDenseVector)
+
+      val adjustedValues = mutable.ArrayBuilder.make[Double]
+      var state: optimizer.State = null
+      while (states.hasNext) {
+        state = states.next()
+        adjustedValues += state.adjustedValue
+      }
+
+      bcFeaturesStd.destroy(blocking = false)
+      if (state == null) {
+        val msg = s"${optimizer.getClass.getName} failed."
+        logError(msg)
+        throw new SparkException(msg)
+      }
+
+      /*
+         The coefficients are trained in the scaled space; we're converting them back to
+         the original space.
+         Note that the intercept in scaled space and original space is the same;
+         as a result, no scaling is needed.
+       */
+      val rawCoefficients = state.x.toArray.clone()
+      val coefficientArray = Array.tabulate(numFeatures) { i =>
+        // flatIndex will loop though rawCoefficients, and skip the intercept terms.
+        val flatIndex = if ($(fitIntercept)) i + i / numFeatures else i
+        val featureIndex = i % numFeatures
+        if (featuresStd(featureIndex) != 0.0) {
+          rawCoefficients(flatIndex) / featuresStd(featureIndex)
+        } else {
+          0.0
+        }
+      }
+
+      val intercept = if ($(fitIntercept)) {
+        rawCoefficients(numFeaturesPlusIntercept - 1)
+      } else {
+        0.0
+      }
+      (Vectors.dense(coefficientArray), intercept, adjustedValues.result())
+    }
+
+    val model = copyValues(new LinearSVCModel(uid, coefficientMatrix, interceptVector))
+    instr.logSuccess(model)
+    model
+  }
+}
+
+@Since("2.2.0")
+object LinearSVC extends DefaultParamsReadable[LinearSVC] {
+
+  @Since("2.2.0")
+  override def load(path: String): LinearSVC = super.load(path)
+}
+
+/**
+ * :: Experimental ::
+ * SVM Model trained by [[LinearSVC]]
+ */
+@Since("2.2.0")
+@Experimental
+class LinearSVCModel private[ml](
+    @Since("2.2.0") override val uid: String,
+    @Since("2.2.0") val coefficients: Vector,
+    @Since("2.2.0") val intercept: Double)
+  extends ClassificationModel[Vector, LinearSVCModel]
+  with LinearSVCParams with MLWritable {
+
+  override val numClasses = 2
+
+  @Since("2.2.0")
+  def setThreshold(value: Double): this.type = set(threshold, value)
```

Comment:
True...not useful yet.  It will be in the future if we add an evaluate() method to return a summary for a new dataset, but it's better not to have it now.